mongodb/trigger/eventlistener/trigger.go

```diff
@@ -0,0 +1,266 @@
+type TriggerFactory struct {
+	metadata *trigger.Metadata
+}
+
+//NewFactory create a new Trigger factory
+func NewFactory(md *trigger.Metadata) trigger.Factory {
+	return &TriggerFactory{metadata: md}
+}
+
+//New Creates a new trigger instance for a given id
+func (t *TriggerFactory) New(config *trigger.Config) (trigger.Trigger, error) {
+	settings := &Settings{}
+	err := metadata.MapToStruct(config.Settings, settings, true)
+	if err != nil {
+		return nil, err
+	}
+	if settings.Connection != "" {
+		mConn, toConnerr := coerce.ToConnection(settings.Connection)
+		if toConnerr != nil {
+			return nil, toConnerr
+		}
+		mclient := mConn.(*mongodb.MongodbSharedConfigManager).GetClient()
+		return &Trigger{metadata: t.metadata, settings: settings, id: config.Id, mclient: mclient}, nil
+	}
+	return nil, nil
+}
+
+// Metadata implements trigger.Factory.Metadata
+func (*TriggerFactory) Metadata() *trigger.Metadata {
+	return triggerMd
+}
+
+// Trigger is a stub for your Trigger implementation
+type Trigger struct {
+	metadata  *trigger.Metadata
+	settings  *Settings
+	evntLsnrs []*EventListener
+	mclient   *mongo.Client
+	logger    log.Logger
+	id        string
+}
+
+// EventListener is structure of a single EventListener
+type EventListener struct {
+	handler      trigger.Handler
+	collName     string
+	listenInsert bool
+	listenUpdate bool
+	listenRemove bool
+	mclient      *mongo.Client
+	database     string
+	done         chan bool
+	logger       log.Logger
+}
+
+// Initialize Mongodb Event Listener
+func (t *Trigger) Initialize(ctx trigger.InitContext) error {
+	t.logger = log.ChildLogger(ctx.Logger(), "mongodb-event-listener")
+	t.logger.Infof("============initializing event listener==")
+	mConn, toConnerr := coerce.ToConnection(t.settings.Connection)
+	if toConnerr != nil {
+		return nil
+	}
+	config := mConn.(*mongodb.MongodbSharedConfigManager).GetClientConfiguration()
+	for _, handler := range ctx.GetHandlers() {
+		s := &HandlerSettings{}
+		err := metadata.MapToStruct(handler.Settings(), s, true)
+		if err != nil {
+			return err
+		}
+		evntLsnr := &EventListener{}
+		evntLsnr.handler = handler
+		evntLsnr.logger = t.logger
+		evntLsnr.collName = s.Collection
+		evntLsnr.listenInsert = s.ListenInsert
+		evntLsnr.listenUpdate = s.ListenUpdate
+		evntLsnr.listenRemove = s.ListenRemove
+		evntLsnr.mclient = t.mclient
+		evntLsnr.database = config.Database
+		evntLsnr.done = make(chan bool)
+		t.evntLsnrs = append(t.evntLsnrs, evntLsnr)
+		t.logger.Debugf("============collName=== %s", evntLsnr.collName)
+		t.logger.Debugf("========listenInsert=== %b", evntLsnr.listenInsert)
+
+	}
+
+	return nil
+}
+
+// Metadata implements trigger.Trigger.Metadata
+func (t *Trigger) Metadata() *trigger.Metadata {
+	return t.metadata
+}
+
+// Start implements trigger.Trigger.Start
+func (t *Trigger) Start() error {
+	t.logger.Infof("Starting Trigger - %s", t.id)
+	for _, evntLsnr := range t.evntLsnrs {
+		pipeline := mongo.Pipeline{}
+		eventOption := 0
```

Comment:
Can you explain your thought process with this? There is else if statement and a switch. They are similar. Also I don't think you'll need "evntLsnr.listenInsert == true" as eventLsnr.listenInsert is already boolean . So you can just have else if and if statement like  "if evntLsnr.listenInsert && !evntLsnr.listenRemove && !vntLsnr.listenUpdate"

Reply:
> Also I don't think you'll need "evntLsnr.listenInsert == true" as eventLsnr.listenInsert is already boolean . So you can just have else if and if statement like if evntLsnr.listenInsert && !evntLsnr.listenRemove && !vntLsnr.listenUpdate

I agree with this comment. Will make the necessary changes.

Reply:
>Can you explain your thought process with this? There is else if statement and a switch. They are similar. 
Since the conditional statement and the object assignment for pipeline object is very verbose, it would be good to handle them separately. It makes the code more readable for other developers. It is redundant I agree. I don't mind having the assignment for pipeline object done inside the conditional statements.